File: 2025-04-April-GFG-POTD/April 02 - BFS of graph/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> bfs(vector<vector<int>>& adj) {
        vector<int> result;
        queue<int> q;
        vector<bool> visited(adj.size(), false);

        q.push(0);  // Start from node 0
        visited[0] = true;

        while (!q.empty()) {
            int node = q.front();
            q.pop();
            result.push_back(node);

            for (int neighbor : adj[node]) {
                if (!visited[neighbor]) {
                    q.push(neighbor);
                    visited[neighbor] = true;
                }
            }
        }
        return result;
    }
};

File: 2025-04-April-GFG-POTD/April 02 - BFS of graph/Solution.py
from collections import deque

class Solution:
    def bfs(self, adj):
        result = []
        queue = deque([0])  # Start from node 0
        visited = [False] * len(adj)
        visited[0] = True

        while queue:
            node = queue.popleft()
            result.append(node)

            for neighbor in adj[node]:
                if not visited[neighbor]:
                    queue.append(neighbor)
                    visited[neighbor] = True
        
        return result
